### ISSUE TITLE
[New Feature] Indicator of number of other rooms with unread messages

### ISSUE THREAD

**Issue by bizzbyster** [RocketChat/Rocket.Chat.Android#1787]
## Description

In many chat applications, one is given an indication of other "rooms" that have unread messages inside a given "room" like the "4" to the right of the back arrow in this screenshot:

<img src="https://user-images.githubusercontent.com/4001792/47269181-64b82d00-d528-11e8-8710-ce91d904c671.png" alt="drawing" width="200"/>

I recently tried to get my daughter to use Rocket.chat Android and she said she couldn't switch b/c it didn't have this feature -- not sure it is so crucial for everyone but that is one user's experience.

**Comment by rafaelks** [comment]
@thiagosanchz Adding Thiago here, I also like this (iOS has it).

**Comment by bizzbyster** [comment]
@rafaelks can you provide a note or two on how this should be implemented to help simplify this for a developer who might want to do the work?

**Comment by ubarsaiyan** [comment]
![screenshot_20181211-043756__01__01](https://user-images.githubusercontent.com/22556377/49767516-cb72ef00-fcfe-11e8-90fb-9dae6707797c.jpg)
@bizzbyster This is how it looks now. I'll try to move the badge a little left. Any other suggestions?

**Comment by ubarsaiyan** [comment]
Code is @ https://github.com/ubarsaiyan/Rocket.Chat.Android/tree/notif-count

**Comment by thiagosanchz** [comment]
I think if you show all unread message on the header of the room will be a lot of distraction to the users.
We created this solution:
![chat pattern - date and new message](https://user-images.githubusercontent.com/39275608/49889303-51d72000-fe28-11e8-9753-26a27b3cab22.png)

**Comment by bizzbyster** [comment]
@thiagosanchz what happens when you click that FAB? My intuition is that you would scroll to the bottom of the room but not go back to the list of rooms.

**Comment by ubarsaiyan** [comment]
@thiagosanchz The FAB button would only notify the users of the new messages in the current room but the unread badge in the top will notify the users of new messages in other rooms. I think we can make it more subtle if it is distracting.

**Comment by ubarsaiyan** [comment]
Different UI implementations:

![2018_12_15_23_19_53](https://user-images.githubusercontent.com/22556377/50701564-3a303780-1074-11e9-9b0a-54c8315f681d.gif)
![screenshot_20181217-224821](https://user-images.githubusercontent.com/22556377/50701582-461bf980-1074-11e9-98b6-c98e0d568259.jpg)
![screenshot_20181217-231012](https://user-images.githubusercontent.com/22556377/50701584-474d2680-1074-11e9-9066-bd224c360287.jpg)
![screenshot_20181217-225108](https://user-images.githubusercontent.com/22556377/50701596-50d68e80-1074-11e9-92ae-af540b55e36f.jpg)

**Comment by ubarsaiyan** [comment]
I prefer the first one (rounded square with number)

**Comment by rafaelks** [comment]
@thiagosanchz @filipedelimabrito Please take a look on these ideas ^^

**Comment by thiagosanchz** [comment]
I agree with @ubarsaiyan The first idea with the number inside a square makes more sense. But I'm still worried about users that have a lot of unread messages, will be a big distraction. IMHO just a circle on the navbar represents nothing.

**Comment by ubarsaiyan** [comment]
@thiagosanchz Maybe per room notification settings (a high priority feature IMHO) like in the web client, can solve the problem of overwhelming the user with many unread rooms. Users can turn off the counter for rooms like #general and #sandbox and won't see the unread badge in the home screen as well as in this unread badge in the top bar. Can anyone think of other ideas?
